### PR TITLE
WIP: Carousel Component improvement

### PR DESCRIPTION
**Pull request by ychek**
Guild Week Task by @ychek & @tombenezra :
Improving the Carousel component

### 🔦 Summary

We are improving the Carousel component of the library to support image slideshow, support custom content which is not images, support swipe behavior for mobile and several more.
We use react-slick library under the hood.

### ✅ Checklist
- [ ] 👨‍💻 API change is approved by the UI Infra Developers 
- [ ] 👨‍🎨 UX change is approved by the Design System UX 
- 📚 Change is documented
  - [ ] Story
  - [ ] API description
  - [ ] Other (explain)
- 🔬 Change is tested
  - [ ] Component
  - [ ] Visual test

### CODE REVIEW

**Comment by shlomitc** [comment]
image
Initial rendering of an unloaded component looks strange, it should maintain the aspect ratio

**Comment by shlomitc** [comment]
Buttons are overflowing
image

File: src/Carousel/Carousel.spec.js [hunk 1]
```diff
@@ -84,12 +89,14 @@ describe('Carousel', () => {
         );
         driver.clickNext();
         expect(driver.getCurrentImageIndex()).toBe(1);
-        jest.runOnlyPendingTimers();
-        driver.clickNext();
-        expect(driver.getCurrentImageIndex()).toBe(2);
-        jest.runOnlyPendingTimers();
-        driver.clickNext();
-        expect(driver.getCurrentImageIndex()).toBe(0);
+        setTimeout(() => {
```
Comment (shlomitc) [inline src/Carousel/Carousel.spec.js]: instead of timeout, use `eventually`

File: src/Carousel/Pagination/Pagination.scss [hunk 2]
```diff
@@ -1,18 +1,22 @@
...
+    border-radius: 10px;
+    margin: 0 6px;
+
+     &:hover {
+      background-color: $B27;
+    }
 
-  &[data-active="true"] {
-    background-color: $B10;
+     & button {
```
Comment (shlomitc) [inline src/Carousel/Pagination/Pagination.scss]: we never style tag elements, only classes

File: test/jest-setup.js [hunk 3]
```diff
@@ -1,7 +1,28 @@
...
+  window.matchMedia ||
+  function() {
+    return {
+      matches: false,
+      addListener: function() {},
+      removeListener: function() {},
+    };
+  };
+
+window.requestAnimationFrame =
```
Comment (shlomitc) [inline test/jest-setup.js]: we have already polyfills folder with RAF, please use it in your test

File: test/jest-setup.js [hunk 4]
```diff
@@ -1,7 +1,28 @@
...
+import 'core-js/es6/set';
+import 'raf/polyfill';
+
 import Enzyme from 'enzyme';
 import Adapter from 'enzyme-adapter-react-15';
 
 Enzyme.configure({ adapter: new Adapter() });
 
+// Required setup for the Carousel test suites to pass because using react-slick
+window.matchMedia =
```
Comment (shlomitc) [inline test/jest-setup.js]: please add this as a polyfill

File: test/jest-setup.js [hunk 5]
```diff
@@ -1,7 +1,28 @@
+import 'core-js/es6/map';
```
Comment (shlomitc) [inline test/jest-setup.js]: why is that?

File: test/jest-setup.js [hunk 6]
```diff
@@ -1,7 +1,28 @@
...
+import 'core-js/es6/set';
+import 'raf/polyfill';
```
Comment (shlomitc) [inline test/jest-setup.js]: do it only in the test file itself

**Comment by ychek** [comment]
@shlomitc do our work will see the light? When we will be able to merge it?

**Comment by shlomitc** [comment]
@ychek - hey, looks like there's some work still left to do as eyes keep failing for visual breaking changes.
@nitayneeman from my team will continue your work.

Thanks for the hard work!

**Comment by nitayneeman** [comment]
@shlomitc I think it's a good point with merge this PR, after that:
* All the component tests are fixed.
* The component is able to receive nodes as children.
* The issue of the storybook example in FF is fixed.
* The component doesn't exceed its container anymore.
* The bottom indicator is aligned to the design system.

image

Notice that two visual and a lot of E2E tests are broken because of running with different versions of Chrome - but I'm not sure it's related these changes...

**Comment by shlomitc** [comment]
Smaller PRs got merged, so I think we can close this one.
Thanks for the hard work everyone

**Pull request closed without merging**